onnxruntime/core/session/ep_library_internal.cc

```diff
@@ -60,8 +60,17 @@ std::unique_ptr<EpLibraryInternal> EpLibraryInternal::CreateCpuEp() {
     return nullptr;
   };
 
+  const auto validate_compatibility_info = [](OrtEpFactory* /*factory*/,
+                                       const char* /*compatibility_info*/,
+                                       OrtCompiledModelCompatibility* model_compatibility) -> OrtStatus* {
+    // CPU EP does not support compiled models. Set the compatibility to OrtCompiledModelCompatibility_SUPPORT_UNKNOWN
```

Comment:
The compatibility validation API, best as I can tell, needs to be plumbed across the internal factory implementation (even to provide a basic implementation), but then this results in all of the provider factories having to provide those based on how the scaffolding is set up (I would need to add the DML and WebGPU ones to this same file, and then we have the other EP factories to contend with). 

Did I miss a simpler way to contend with this? Am I going in the wrong direction here? cc: @adrianlizarraga @skottmckay